### PR TITLE
fix broken tests

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Fixes two issues:

1. Broken crash handler tests
   These tests were broken since the general test verbosity was reduced in devel a few months ago. these tests check for specific log messages, which were not emitted after the log verbosity was reduced. Now the tests specifically enable the log topic in question, so that all required log messages for the tests are emitted again.

2. Adjust HTTP response code for error number 1521 from 500 to 400
   Error 1521 (query collection lock failed) is nowadays only emitted in traversals, when a collection is accessed during the traversal that has not been specified in the WITH statement of the query. Thus returning HTTP 500 is not a good idea, as it is clear a user error that triggered the problem.

No backports are planned for this.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: -
  - [ ] Backport for 3.8: -
  - [ ] Backport for 3.7: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Review by cpjulia — APPROVED** [review]
LGTM

**Review by neunhoef — APPROVED** [review]
LGTM. Heretic question: How can it be that these missing log messages break tests but we never saw test failures since the logging changes? Or are these tests not run in PR or nightly tests?

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,14 @@
 devel
 -----
 
+* Changed HTTP response code for error number 1521 from 500 to 400.
```
[the remark this answers is not in the document]
Comment (jsteemann) [reply CHANGELOG:4]: The test fix is an internal-only change that has no effect on end users. Thus it is not mentioned in the CHANGELOG to reduce clutter.

**Comment by jsteemann** [comment]
IMO it is definitely not easier to do this in the testing framework.

**Comment by dothebart** [comment]
https://github.com/arangodb/arangodb/blob/devel/js/client/modules/%40arangodb/testsuites/recovery.js#L64
add
```
'log.level': 'crash=info'
```
done.

**Pull request merged**